artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -1136,7 +1136,16 @@ private void sendContinuations(final int packetSize,
             currentLargeMessage = null;
 
             try {
-               session.send(session.getCurrentTransaction(), EmbedMessageUtil.extractEmbedded((ICoreMessage) message.toMessage(), storageManager), false, producers.get(senderID), false);
+               Message m = EmbedMessageUtil.extractEmbedded((ICoreMessage) message.toMessage(), storageManager);
```

Comment:
shouldn't the fix be to actually keep the message as large instead?

say, if you look here: 
https://github.com/apache/activemq-artemis/blob/257dd86ae28205d8ad57314cbe1cf8b17659aea5/artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/protocol/EmbedMessageUtil.java#L78-L79

we could check on EMBEDDED_TYPE:


if (message.isLargeMessage) then perform the extractLargeMessage


or the encoded version wouldn't be compatible in that case?